src/commands/math.ts

```diff
@@ -330,100 +355,122 @@ var MathCommand = P(MathElement, function(_, super_) {
+      text = ctrlSeq.replace(/^\\/, '');
+    }
 
     this.mathspeakName = mathspeak || text;
-    super_.init.call(this, ctrlSeq, html, [ text ]);
-  };
+    super.setCtrlSeqHtmlAndText(ctrlSeq, html, [text || '']);  
+  }
 
-  _.parser = function() { return Parser.succeed(this); };
-  _.numBlocks = function() { return 0; };
+  parser () { return Parser.succeed(this); };
+  numBlocks () { return 0; };
 
-  _.replaces = function(replacedFragment) {
+  replaces (replacedFragment:Fragment) {
     replacedFragment.remove();
   };
-  _.createBlocks = noop;
+  createBlocks () {};
 
-  _.moveTowards = function(dir, cursor) {
+  moveTowards (dir:Direction, cursor:Cursor) {
     cursor.jQ.insDirOf(dir, this.jQ);
-    cursor[-dir] = this;
+    cursor[-dir as Direction] = this;
     cursor[dir] = this[dir];
     aria.queue(this);
   };
-  _.deleteTowards = function(dir, cursor) {
+  deleteTowards (dir:Direction, cursor:Cursor) {
     cursor[dir] = this.remove()[dir];
   };
-  _.seek = function(pageX, cursor) {
+  seek (pageX:number, cursor:Cursor) {
     // insert at whichever side the click was closer to
     if (pageX - this.jQ.offset().left < this.jQ.outerWidth()/2)
       cursor.insLeftOf(this);
     else
       cursor.insRightOf(this);
-  };
 
-  _.latex = function(){ return this.ctrlSeq; };
-  _.text = function(){ return this.textTemplate.join(''); };
-  _.mathspeak = function(){ return this.mathspeakName; };
-  _.placeCursor = noop;
-  _.isEmpty = function(){ return true; };
-});
-var VanillaSymbol = P(Symbol, function(_, super_) {
-  _.init = function(ch, html, mathspeak) {
-    super_.init.call(this, ch, '<span>'+(html || ch)+'</span>', undefined, mathspeak);
+    return cursor;
   };
-});
-var BinaryOperator = P(Symbol, function(_, super_) {
-  _.init = function(ctrlSeq, html, text, mathspeak) {
-    super_.init.call(this,
-      ctrlSeq, '<span class="mq-binary-operator">'+html+'</span>', text, mathspeak
-    );
+
+  latex (){ return this.ctrlSeq || ''; };
+  text (){ return this.textTemplate.join(''); };
+  mathspeak (_opts?:MathspeakOptions){ return this.mathspeakName || ''; };
+  placeCursor () {};
+  isEmpty (){ return true; };
+};
+class VanillaSymbol extends MQSymbol {
+  constructor (ch:string, html?:string, mathspeak?:string) {
+    super(ch, '<span>'+(html || ch)+'</span>', undefined, mathspeak);
+  };
+}
+function bindVanillaSymbol (ch:string, html?:string, mathspeak?:string) {
+  return () => new VanillaSymbol(ch, html, mathspeak);
+}
+
+class BinaryOperator extends MQSymbol {
+  constructor (ctrlSeq?:string, html?:string, text?:string, mathspeak?:string, treatLikeSymbol?:boolean) {
+    if (treatLikeSymbol) {
+      super(ctrlSeq, '<span>'+(html || ctrlSeq)+'</span>', undefined, mathspeak);
+    } else {
+      super(ctrlSeq, '<span class="mq-binary-operator">'+html+'</span>', text, mathspeak);
+    }
   };
-});
+};
+function bindBinaryOperator (ctrlSeq?:string, html?:string, text?:string, mathspeak?:string) {
```

Comment:
I think this function could maybe use a better name now that we are getting rid of the generic unsafe “bind” concept